ClosedXML/Excel/Protection/IXLProtectable.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Keep this file CodeMaid organised and cleaned
+using System;
+using static ClosedXML.Excel.XLProtectionAlgorithm;
+
+namespace ClosedXML.Excel
+{
+    public interface IXLProtectable<TProtection, in TElement> : IXLProtectable
+        where TProtection : IXLElementProtection<TElement>
+        where TElement : struct
+    {
+        TProtection Protect();
+
+        TProtection Protect(String password, Algorithm algorithm = DefaultProtectionAlgorithm);
+
+        TProtection Protect(String password, Algorithm algorithm, TElement allowedElements);
+
+        TProtection Protection { get; set; }
+
+        TProtection Unprotect();
+
+        TProtection Unprotect(String password);
+    }
+
+    public interface IXLProtectable
+    {
+    }
```

Comment:
You probably will want to kill me but I suggest adding non-generic members here:

```
bool IsProtected { get; }

IXLElementProtection Protect();

IXLElementProtection Protect(String password, Algorithm algorithm = DefaultProtectionAlgorithm);

IXLElementProtection Unprotect();

IXLElementProtection Unprotect(String password);
```

Reply:
But then method chaining for the enums will break, right? `ws.Protect().AllowedElements = value`

Reply:
No, `IXLWorksheet` will still be `IXLProtectable<XLSheetProtection>` and it will keep working. By introducing these non-generic members we'll enable something like this:

```
var list = new List<IXLProtectable>{ wb, ws1, ws2 };
list.Where(el => !el.IsProtected).ForEach(el => el.Protect());
list.Where(el => el.IsProtected).ForEach(el => el.Unprotect());
```

Reply:
Oh, I see now you're suggesting that I add them to the non-generic interface. I first thought you wanted me to change the signatures of the methods in the generic class.

Reply:
Done